# Space Mouse Configuration for TeleopTwistStampedJoy
teleop_joy_to_wrench:
  ros__parameters:
    # Publishing rate (Hz)
    publish_rate: 50.0
    
    # Scaling factors
    force_scale: 1.0      # Max linear velocity (m/s)
    torque_scale: 1.0     # Max angular velocity (rad/s)
    
    x_axis: 0         # Camera pan left/right (Stick X)
    y_axis: 1         # Camera pan up/down (Stick Z)
    z_axis_up: 5         # Camera push (Stick Y)
    z_axis_down: 2         # Camera pull (Stick Y)
    roll_axis_positive: 3        # Camera roll along positive Z-axis
    roll_axis_negative: 0         # Camera roll along negative Z-axis
    pitch_axis: 3       # Camera pitch along X-axis
    yaw_axis: 4         # Camera yaw along Y-axis
    enable_button: -1       # Safety button to enable/disable control (-1 fr no safety button)

    invert_x: true    # Invert X-axis (left/right)
    invert_y: false   # Invert Y-axis (up/down)
    invert_z: false   # Invert Z-axis (push/pull)
            
    # Deadzone to prevent drift (Xbox controllers typically need higher deadzone)
    deadzone: 0.05
    
    # Frame ID for the twist messages
    frame_id: "eoat_camera_link"
    
    # Topic names
    joy_topic: "joy"
    twist_topic: "teleop_force_threshold"  # Updated topic for inspection cell